flax/linen/fp8_ops.py

```diff
@@ -645,17 +603,18 @@ def __call__(self, *args, **kwargs):
       *args, **kwargs
     )
 
-    y = q_dot_dq(
+    y = fp8_scaled_dot_general(
```

Comment:
fp8_scaled_dot_general implies direct quantization by design, so consider change name class Fp8DirectDotGeneralOp to class Fp8DotGeneralOp and the old class Fp8DotGeneralOp to Fp8FakeDotGeneralOp to be more clear. Also consider marking the fake dot general op to be deprecated.

Reply:
I've added a deprecation warning, but kept the existing API names unchanged to avoid breaking user code.